### PR TITLE
Remove pcmk__cli_option_t and associated code

### PR DESCRIPTION
**Pull request by clumens**

### CODE REVIEW

**Review by kgaillot — COMMENTED** [review]
Woohoo! Very satisfying :)

File: include/crm/common/options_internal.h [hunk 1]
```diff
@@ -24,46 +24,7 @@
  * This will all eventually go away as everything is converted to use GOption
  */
```
[the remark this answers is not in the document]
Comment (clumens) [reply include/crm/common/options_internal.h:25]: Removed.

**Comment by nrwahl2** [comment]
Hooray!

File: include/crm/common/options_internal.h [hunk 2]
```diff
@@ -21,49 +21,9 @@
 /*
  * Command-line option handling
  *
- * This will all eventually go away as everything is converted to use GOption
  */
```
Comment (kgaillot) [inline include/crm/common/options_internal.h:24]: Actually the entire comment is no longer relevant

Comment (clumens) [reply include/crm/common/options_internal.h]: Nuked the whole thing.

**Review by kgaillot — APPROVED** [review]
Ah, I overlooked that pcmk__cli_help() was still in that section. It's fine as is though

**Pull request merged**